### PR TITLE
Refresh vehicle positions

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR enables refreshing vehicle positions at a configured interval (default 30s) when the route viewer is visible.
A new config parameter `RouteViewer > vehiclePositionRefreshSeconds` handles that.

Remarks:
- That functionality can't be disabled at this time, let me know if it should.

**PR Checklist:**
- [na] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [na] Are all languages supported (Internationalization/Localization)?
- [na] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks great! My only concern is that `getVehiclePositionsForRoute` automatically resets the map center and zoom. If you are zooming in to more closely investigate a route with realtime vehicles, I think it could be annoying and confusing to suddenly have the map jump back to its original position with no indication why. Maybe there's a way to refresh vehicle positions without the map movement?

**Comment by binh-dam-ibigroup** [comment]
> Looks great! My only concern is that `getVehiclePositionsForRoute` automatically resets the map center and zoom. If you are zooming in to more closely investigate a route with realtime vehicles, I think it could be annoying and confusing to suddenly have the map jump back to its original position with no indication why. Maybe there's a way to refresh vehicle positions without the map movement?

Excellent point, let me see if the map refitting requires a patch in OTP-UI.

**Comment by binh-dam-ibigroup** [comment]
> > Looks great! My only concern is that `getVehiclePositionsForRoute` automatically resets the map center and zoom. If you are zooming in to more closely investigate a route with realtime vehicles, I think it could be annoying and confusing to suddenly have the map jump back to its original position with no indication why. Maybe there's a way to refresh vehicle positions without the map movement?
> 
> Excellent point, let me see if the map refitting requires a patch in OTP-UI.

@amy-corson-ibigroup This is now fixed in https://github.com/opentripplanner/otp-react-redux/pull/832/commits/cf8e07587e52b3027bb217d0623f74ca76e92593, and no changes are needed to OTP-UI!

File: lib/components/viewers/route-details.js [hunk 1]
```diff
@@ -42,13 +42,42 @@ class RouteDetails extends Component {
...
     route: PropTypes.shape({ id: PropTypes.string }),
     setHoveredStop: setHoveredStop.type,
-    setViewedRoute: setViewedRoute.type
+    setViewedRoute: setViewedRoute.type,
+    vehiclePositionRefreshSeconds: PropTypes.number
   }
 
-  componentDidMount = () => {
+  state = {
+    refreshTimer: null
```
Comment (miles-grant-ibigroup) [inline lib/components/viewers/route-details.js:50]: This should probably be named something more explicit!
```suggestion
    vehiclePositionsRefreshTimer: null
```

**Review by miles-grant-ibigroup — APPROVED** [review]
Works great! This is a long-overdue feature and this is a very clean implementation. Thanks very much!

File: lib/components/map/connected-route-viewer-overlay.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    return `${routeId}-${patternId || ''}`
+  }
+)
+
+// connect to the redux store
+
+const mapStateToProps = (state: any) => {
+  const { patterns = {}, pending } = extractRoute(state)
+  // If the state is not pending and filtered patterns contains at least one key,
+  // memoize the route data (based on route id and pattern id)
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/map/connected-route-viewer-overlay.ts]: Actually we don't. Had been thinking about it over the weekend... `memoize` would retain every combination of routes/patterns the user opens, and we don't need that, we just need to cache the route data for the current route/pattern. I have simplified in https://github.com/opentripplanner/otp-react-redux/pull/832/commits/cd06830307dcc62f51ceb5bc9851e1ceebbab111.

**Pull request merged**